airflow/config_templates/config.yml

```diff
@@ -2246,6 +2246,13 @@
       type: string
       example: ~
       default: "offset"
+    - name: index_patterns
+      description: |
+        Comma separated list of index patterns to use when searching for logs (default: `_all`).
+      version_added: 2.3.0
+      type: string
+      example: something-*
+      default: "_all"
```

Comment:
Is this a good default? Since this handles wildcard expansion, perhaps we could simply make the default `*` instead.

Reply:
@uranusjr indeed we could use `*`. According to the documentation it is equivalent to `_all`. See
https://www.elastic.co/guide/en/elasticsearch/reference/8.2/search-search.html#search-search-api-path-params

Initially I had it set to `*` and then changed it to `_all` because it seemed more readable. That said, I don't mind changing it back to `*` if ASF members prefer it.

Reply:
If this is a special case in Elastic, I am OK to use `_all` as the default. The main motivation behind using `*` is you don’t need a special case in `_validate_search_targets` for it. But if we need to handle `_all` specially anyway, the difference is minimal.

Reply:
For background:
https://github.com/elastic/elasticsearch-py/blob/d66dfcb9c7e2bc592392ca00fde62889d181c0af/elasticsearch/_sync/client/__init__.py

Reply:
@uranusjr I'm not 100% sure but I think that it's a very common Elasticsearch idiom to use `_all` or the empty string. Perhaps we could change the default to an empty string?

Reply:
in fact I i'm realising now that `FakeElasticsearch` should probably handle the empty string case.